diff --git a/src/map/ai/controllers/player_controller.cpp b/src/map/ai/controllers/player_controller.cpp
index 1c0df703fa..0e57d4a212 100644
--- a/src/map/ai/controllers/player_controller.cpp
+++ b/src/map/ai/controllers/player_controller.cpp
@@ -75,18 +75,11 @@ bool CPlayerController::Engage(uint16 targid)
     {
         if (distance(PChar->loc.p, PTarget->loc.p) < 30)
         {
-            if (m_lastAttackTime + std::chrono::milliseconds(PChar->GetWeaponDelay(false)) < server_clock::now())
+            if (CController::Engage(targid))
             {
-                if (CController::Engage(targid))
-                {
-                    PChar->PLatentEffectContainer->CheckLatentsWeaponDraw(true);
-                    PChar->pushPacket(new CLockOnPacket(PChar, PTarget));
-                    return true;
-                }
-            }
-            else
-            {
-                errMsg = std::make_unique<CMessageBasicPacket>(PChar, PTarget, 0, 0, MSGBASIC_WAIT_LONGER);
+                PChar->PLatentEffectContainer->CheckLatentsWeaponDraw(true);
+                PChar->pushPacket(new CLockOnPacket(PChar, PTarget));
+                return true;
             }
         }
         else
